### ISSUE TITLE
Unmaximizing moves tag to different screen

### ISSUE THREAD

**Issue by dbedrenko** [awesomeWM/awesome#1615]
Output of `awesome --version`:
```
awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.3.3 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✔
 • LGI version: 0.9.1
```

**How to reproduce the issue:**

0. Have 2 screens with different sizes (because I suspect it could be related to this problem). Autofocus should be enabled in your rc.lua and sloppy focus should be disabled.
1. Switch the tag layout on both screens to "max".
2.  Start a GUI program on 1st screen. I tested with Konsole and VLC.
3. Toggle maximize on the client.
4. Move the client to the 2nd screen using the keyboard shortcut.
5. Toggle maximize on the client.

**Actual result:**

Client jumps back to the first 1st screen.

**Expected result:**

Client should stay on the screen it was on.

**Comment by Elv13** [comment]
Fixed in master, see #1320

**Comment by dbedrenko** [comment]
Should've searched for closed issues too, sorry about that.
Thanks for your work!
